WordPress/Docs/WP/TimezoneChangeStandard.xml

```diff
@@ -0,0 +1,22 @@
+<documentation title="Disallow the changing of timezone.">
+    <standard>
+    <![CDATA[
+      Using date_default_timezone_set() and similar isn't allowed, instead use WP internal timezone support.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: Using WP internal timezone support.">
```

Comment:
The description doesn't match up with the example. The description says WP, but the example is pure PHP!

(Pulled from the unit test file.)

Reply:
I've done some tracing back of this and it looks like the sniff has not had any significant changes (other than sniff technical improvements) since it was first added to the library:
* Issue #76 requested it.
* PR #103 added it.

The issue and the inline docs refer to the following links as relevant:
* https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/#manipulating-the-timezone-server-side
* https://wpvip.com/documentation/vip-go/use-current_time-not-date_default_timezone_set/
* https://weston.ruter.net/2013/04/02/do-not-change-the-default-timezone-from-utc-in-wordpress/

Based on the information there, the more recent issue around the use of `gmdate()` instead of `date()` - #1713 - is very much related to this as well.

Now there are two things which are confusing here:
1. The unit test "good" case.
    That case shouldn't really be there as it doesn't test the sniff at all.
    The sniff is not looking for anything in the code of the "good" case, so would never trigger on it.
    It could be replaced with something like `My_Class::date_default_timezone_set( 'Foo/Bar' );` which would be testing the proper functioning of the sniff against false positives.
2. The error message refers to "_WP internal timezone support_" which is an unclear phrase to begin with.
    This is IMO where the docs you are creating come in and can help hugely to clarify what is meant by this, though if someone can come up with a simple phrase to improve the error message, that could be helpful too.
    I think that summarizing the information found in the last two docs links I mention above, to come up with a better description and better code samples would be the way forward.
    I can also imagine some additional issues may need to be opened, sooner rather than later, to get WPCS to detect additional cases of _doing it wrong_ regarding retrieving/displaying date/time information. /cc @Rarst 

On a related note - PR #1714 added a check for the use of `date()` to the `WordPress.PHP.RestrictedPHPFunctions` sniff.
This change has not yet been released and based on the above, I'm now wondering if we should move that check to this sniff instead.

What do you think ?

Reply:
Yeah, `gmdate()` change is purely to make timezone issue more foolproof, so if it can be here.

> Using date_default_timezone_set() and similar isn't allowed, instead use WP internal timezone support.

Now this line is a mess. There is no "similar", there aren't other things that would mess up default time zone. And this doesn't really come up in _WordPress_ context, this comes up when people do _PHP_ in WordPress in incompatible way.

It should be something like this (probably in sniff itself as well?):

> Using date_default_timezone_set() isn't allowed, because WP core needs default time zone set to UTC. Use DateTime object or core API functions instead.

PS https://wpvip.com/documentation/vip-go/use-current_time-not-date_default_timezone_set/ this is highly outdated rec, local timestamps are a huge problem (I am close to getting rid of `current_time('timestamp')` in core) and `timezone_string` is highly preferable to `gmt_offset` if available (that mess is about to get abstracted as `wp_timezone()` in core).